src/main/java/org/broadinstitute/hellbender/utils/variant/GATKVariantContextUtils.java

```diff
@@ -40,40 +42,69 @@ public static boolean isInformative(final double[] gls) {
     }
 
     /**
-     * Creates a VariantContextWriter who's outputFile type is <code>VariantContextWriterBuilder.OutputType.VCF</code>.
+     * Creates a VariantContextWriter who's outputFile type is based on the extension of the output file name.
      * The default options set by VariantContextWriter are cleared before applying ALLOW_MISSING_FIELDS_IN_HEADER (if
      * <code>lenientProcessing</code> is set), followed by the set of options specified by any <code>options</code> args.
      *
-     * @param outFile output File for this writer
+     * @param outFile output File for this writer. May not be null.
      * @param referenceDictionary required if on the fly indexing is set, otherwise can be null
-     * @param lenientProcessing true if ALLOW_MISSING_FIELDS_IN_HEADER option should be set
-     * @param options variable length list of additional options to be set for this writer
+     * @param createMD5 true if an md5 file should be created
+     * @param options variable length list of additional Options to be set for this writer
      * @returns VariantContextWriter must be closed by the caller
      */
     public static VariantContextWriter createVCFWriter(
```

Comment:
Is that something we want this method to do?  We'd have to add additional parameters to this method to take the gqPartitions and default ploidy.  I'm not sure any tools that aren't haplotype caller would want to write a gvcf, so I think it would be ok if it just does the wrapping of the output writer manually still?